pkg/iac-providers/cft/v1/load-file.go

```diff
@@ -89,39 +90,112 @@ func (a *CFTV1) getConfig(absFilePath string, fileData *[]byte, parameters *map[
 }
 
 func (a *CFTV1) extractTemplate(file string, data *[]byte) (*cloudformation.Template, error) {
+	var multiErr multierror.Error
 	fileExt := a.getFileType(file, data)
+	var isYaml bool
 
 	switch fileExt {
 	case YAMLExtension, YAMLExtension2:
-		zap.S().Debug("sanitizing cft template file", zap.String("file", file))
-		sanitized, err := a.sanitizeCftTemplate(*data, true)
-		if err != nil {
-			zap.S().Debug("failed to sanitize cft template file", zap.String("file", file), zap.Error(err))
-			return nil, err
+		isYaml = true
+	case JSONExtension:
+		isYaml = false
+	default:
+		zap.S().Debug("unknown extension found", zap.String("extension", fileExt))
+		return nil, fmt.Errorf("unsupported extension for file %s", file)
+	}
+
+	zap.S().Debug("sanitizing cft template file", zap.String("file", file))
+	sanitized, err := a.sanitizeCftTemplate(*data, isYaml)
+	if err != nil {
+		zap.S().Debug("failed to sanitize cft template file", zap.String("file", file), zap.Error(err))
+		return nil, err
+	}
+
+	preParsedList, err := preParse(sanitized)
+	if err != nil {
+		zap.S().Debug("failed to unmarshal sanitized json", zap.String("file", file), zap.Error(err))
+		return nil, err
+	}
+
+	var onetemplate cloudformation.Template
+	onetemplate.Resources = make(map[string]cloudformation.Resource, 1)
+	for i := range preParsedList {
+		var resourceName string
+		for key := range preParsedList[i].Resources {
+			resourceName = key
 		}
-		template, err := goformation.ParseYAML(sanitized)
+
+		resourceData, err := json.Marshal(preParsedList[i])
 		if err != nil {
-			zap.S().Debug("failed to parse file", zap.String("file", file), zap.Error(err))
-			return nil, err
+			zap.S().Debug("failed to marshal json for resource", zap.String("resource", resourceName), zap.Error(err))
+			multiErr.Errors = append(multiErr.Errors, err)
+			continue
 		}
-		return template, nil
-	case JSONExtension:
-		zap.S().Debug("sanitizing cft template file", zap.String("file", file))
-		sanitized, err := a.sanitizeCftTemplate(*data, false)
+
+		template, err := goformation.ParseJSON(resourceData)
 		if err != nil {
-			zap.S().Debug("failed to sanitize cft template file", zap.String("file", file), zap.Error(err))
-			return nil, err
+			zap.S().Debug("failed to generate template for resource", zap.String("resource", resourceName), zap.Error(err))
+			multiErr.Errors = append(multiErr.Errors, err)
```

Comment:
use `multierr.Append` and append to `a.errIacLoadDirs`. I think all the errors added should be of type `DirScanErr`